AsyncDisplayKit/Private/ASInternalHelpers.h

```diff
@@ -33,6 +33,26 @@ BOOL ASRunningOnOS7();
 ASDISPLAYNODE_EXTERN_C_END
 
 /**
+ @summary Conditionally performs UIView geometry changes in the given block without animation and call completion block afterwards.
+ 
+ Used primarily to circumvent UITableView forcing insertion animations when explicitly told not to via
+ `UITableViewRowAnimationNone`. More info: https://github.com/facebook/AsyncDisplayKit/pull/445
+ 
+ @param withoutAnimation Set to `YES` to perform given block without animation
+ @param block Perform UIView geometry changes within the passed block
+ @param completion Call completion block if UIView geometry changes within the passed block did complete
+ */
+ASDISPLAYNODE_INLINE void ASPerformBlockWithoutAnimationCompletion(BOOL withoutAnimation, void (^block)(), void (^completion)()) {
```

Comment:
If you look at the whole diff you will see the only thing I changed is adding a new function with completion block. Before there already was this kind of function just without the completion block. To be able to use a completion block we have to move from the UIView performWithoutAnimation API to CATransaction based API as we can set the completion block there. So theoretically nothing should have changed.